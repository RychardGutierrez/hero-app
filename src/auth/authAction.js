export const actionContex = (type, payload) => ({
  type,
  payload,
});
