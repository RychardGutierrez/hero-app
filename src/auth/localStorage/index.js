export const setStorage = (nameStorage, data) =>
  localStorage.setItem(nameStorage, JSON.stringify(data));

export const getStorage = (nameStorage) =>
  JSON.parse(localStorage.getItem(nameStorage));
