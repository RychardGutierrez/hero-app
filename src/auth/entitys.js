import { getStorage } from './localStorage';
import { STORAGE_USER } from './localStorage/constants';

export const initReducer = () => {
  const initState = getStorage(STORAGE_USER) || {
    logged: false,
  };

  return initState;
};
