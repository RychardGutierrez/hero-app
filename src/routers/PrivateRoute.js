import { useContext } from 'react';
import { Navigate, useLocation } from 'react-router-dom';

import { AuthContext } from '../auth/authContext';
import { STORAGE_PATH } from '../auth/localStorage/constants';
import { setStorage } from './../auth/localStorage/index';

const PrivateRoute = ({ children }) => {
  const { user } = useContext(AuthContext);

  const { pathname, search } = useLocation();

  setStorage(STORAGE_PATH, `${pathname}${search}`);

  return user.logged ? children : <Navigate to={'/login'} />;
};

export default PrivateRoute;
