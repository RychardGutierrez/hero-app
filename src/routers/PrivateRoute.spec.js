import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../auth/authContext';
import PrivateRoute from './PrivateRoute';

Storage.prototype.setItem = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Navigate: () => <span>Exit component</span>,
}));

describe('PrivateRoute', () => {
  it('should show of the component if user is autenticate and save in the localstorage', () => {
    const contextValue = {
      user: {
        name: 'Lucas',
        logged: true,
      },
      dispatch: jest.fn(),
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/']}>
          <PrivateRoute>
            <h1>Render component</h1>
          </PrivateRoute>
        </MemoryRouter>
      </AuthContext.Provider>,
    );
    expect(wrapper.find('h1').text().trim()).toBe('Render component');
    // expect(localStorage.setItem).toHaveBeenCalledWith('path', "\"/\"");

    expect(wrapper).toMatchSnapshot();
  });

  it('should block component is the user is not authenticate', () => {
    const contextValue = {
      user: {
        logged: false,
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/']}>
          <PrivateRoute>
            <h1>Render component</h1>
          </PrivateRoute>
        </MemoryRouter>
      </AuthContext.Provider>,
    );
    expect(wrapper.find('span').text().trim()).toBe('Exit component');
    // expect(localStorage.setItem).toHaveBeenCalledWith('path', "\"/\"");

    expect(wrapper).toMatchSnapshot();
  });
});
