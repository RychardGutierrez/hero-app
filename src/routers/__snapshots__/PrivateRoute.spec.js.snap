// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should block component is the user is not authenticate 1`] = `
<span>
  Exit component
</span>
`;

exports[`PrivateRoute should show of the component if user is autenticate and save in the localstorage 1`] = `
<h1>
  Render component
</h1>
`;
